.github/workflows/_pydantic_compatibility.yml

```diff
@@ -48,8 +48,8 @@ jobs:
           REGULAR_VERSION=$(poetry run python -c "import pydantic; print(pydantic.__version__)" | cut -d. -f1)
 
           if [[ "$REGULAR_VERSION" == "1" ]]; then
-            PYDANTIC_DEP=">=2.1,<3"
```

Comment:
why are we changing this?

Reply:
I removed the compatibility test for pydantic V2 because I found out during testing that pydantic V2 does not pickle classmethod, but the @root_validator decorator used by our public `BaseModel` returns a class method. So, I'm wondering if compatibility with pydantic V2 is a universal issue.

Reply:
I saw your modification, a very good solution